test/functional/editor/completion_spec.lua

```diff
@@ -1228,4 +1228,39 @@ describe('completion', function()
     expect('colorscheme NOSUCHCOLORSCHEME')
     assert_alive()
   end)
+
+  it('complete with f flag #25598', function()
+    screen:try_resize(20, 9)
+    local bufname = 'foo/bar.txt'
+    local hidden = 'fooA/.hidden'
+    local is_win = helpers.is_os('win')
+    local head = {
+      "foo^                 |",
+      "{2:foo            }{0:     }|",
+    }
+    if is_win then
+      bufname = 'C:\\foo\\bar.txt'
+      hidden = 'C:\\fooA\\.hidden'
+      head = {
+        "C:^                  |",
+        "{2:C:             }{0:     }|",
+        "{1:foo            }{0:     }|",
+      }
+    end
+    command('set complete+=f | edit '.. bufname ..' | edit '..hidden)
+    feed('i<C-n>')
+
+    local grid = {
+      "{1:bar            }{0:     }|",
+      "{1:txt            }{0:     }|",
+      "{1:fooA           }{0:     }|",
+      "{1:.hidden        }{0:     }|",
+      "{0:~                   }|",
+      ("{3:-- }{4:match 1 of %d}     |"):format(is_win and 6 or 5),
+    }
+    if not is_win then
+      table.insert(grid, #grid - 1, "{0:~                   }|")
+    end
+    screen:expect{grid=table.concat(vim.list_extend(head,grid), '\n')}
```

Comment:
I'd suggest just write

```
if is_win then
   screen:expect{grid=...}
else
   screen:expect{grid=...}
end
```
It is fine if screen tests are verbose (non DRY) to make is easier to see what is going on at a quick glance.

Reply:
done .